### ISSUE TITLE
some error when make -j

### ISSUE THREAD

**Issue by pengxutao** [colmap/colmap#1482]
**Describe the bug**
when execute "make -j",some error occur.

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version 18.04
 - gcc and g++ version:7.5.0

**Additional context**
`[ 40%] Building CXX object src/CMakeFiles/colmap.dir/base/camera.cc.o
In file included from /usr/local/include/ceres/ceres.h:67:0,
                 from /home/xuzhi/pengxutao/colmap/src/base/camera_models.h:42,
                 from /home/xuzhi/pengxutao/colmap/src/base/camera.cc:36:
/usr/local/include/ceres/product_manifold.h: In static member function ‘static std::array<T, N> ceres::ProductManifold<Manifold0, Manifold1, ManifoldN>::ExclusiveScan(const std::array<T, N>&)’:
/usr/local/include/ceres/product_manifold.h:260:10: error: ‘exclusive_scan’ is not a member of ‘std’
     std::exclusive_scan(values.begin(), values.end(), result.begin(), 0);
          ^~~~~~~~~~~~~~
src/CMakeFiles/colmap.dir/build.make:62: recipe for target 'src/CMakeFiles/colmap.dir/base/camera.cc.o' failed
make[2]: *** [src/CMakeFiles/colmap.dir/base/camera.cc.o] Error 1
CMakeFiles/Makefile2:917: recipe for target 'src/CMakeFiles/colmap.dir/all' failed
make[1]: *** [src/CMakeFiles/colmap.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
`
I had checked the file "/usr/local/include/ceres/product_manifold.h",and found that it had #include <numeric>,so why there is a error: error: ‘exclusive_scan’ is not a member of ‘std’.

help

**Comment by whuaegeanse** [comment]
@pengxutao 
exclusive_scan is a function of c++17.
Use ceres 2.1 or 2.0 instead of master branch  becasuse ceres requires c++17.
[Ceres Solver now requires C++17](https://github.com/ceres-solver/ceres-solver/commit/1274743609bc59621adc9e311cdeeaad7eb65201)

**Comment by pengxutao** [comment]
@whuaegeanse 
thank you so much! I used your method and install ceres 2.0, then everything is OK! and I had successfully install colmap.Thank you!

**Comment by judafe** [comment]
The comments in this issue helped but were insufficient. I had to take an additional step to successfully install Ceres. 

Ceres v2.0.0 does not build on my machine. Running CMake produced an error at ```cmake/FindTbb.cmake```, as detailed in this [issue](https://github.com/ceres-solver/ceres-solver/issues/669). 

As a fix, I checked out a [commit that patched the error](https://ceres-solver-review.googlesource.com/c/ceres-solver/+/18500).

```bash
git checkout 941ea13475913ef8322584f7401633de9967ccc8
```
I hope this helps anyone who faces the same problem.

P.S. my OS is  Pop!_OS 22.04 LTS
